example/output/ts/ShipDate.ts

```diff
@@ -0,0 +1,6 @@
+export const ShipDate = {
```

Comment:
Why something like this would be useful?
This is a constant object with constant keys and values. Not a typescript type that we can re-use

Reply:
`oas2tson` produces constant objects that represent OpenAPI schemas to support the JSON schema type provider in Fastify. The corresponding TypeScript type in `example/output/types/ShipDate.d.ts` is `export type ShipDate = Date;`.

Your question comes down to choices about how to build an OpenAPI schema. My background is data modeling, metadata, large data warehouses, data management, etc. That leads me to value metadata about data elements so people know what they are and know issues to consider. (For example, consider the differences between a "city" as defined by legal boundaries and a "city" as defined by the post office (may not match) and needing to know which is in play in different data objects.) OpenAPI specs are decent metadata stores as outlined below.

```yaml
# Fields.yaml
components:
  schemas:
    ShipDate:
      description: |-
        A detailed description about ShipDate because you're using your OpenAPI spec as a metadata repository for your application because your company doesn't have one but you have enough experience to know metadata is critical for any data source if you want it to be usable.

        And you like using `markdown` so you can reference [links](https://example.com/blah-blah-blah), include tables, etc., as needed to make API documentation useful for API consumers, not just the afterthought that many internal APIs I see seem to think of it as.
      type: string
      format: date-time
    ...

# Orders.yaml
    Order:
      type: object
      properties:
        shipDate:
          $ref:  'Fields.yaml#/components/schemas/ShipDate'
        ...
        # all properties members ref `Fields.yaml`
    OrderQuery:
      type: object
      properties:
        shipDate:
          $ref: '#/components/schemas/ShipDate'
       # all properties members ref `Fields.yaml`
    ...
# and anywhere else I need ShipDate -- DRY.
```


```typescript
import {ShipDate as ShipDateSchema} from `.../ShipDate.ts`

const schema = {
  params: ShipDateSchema
}

// Arguably, shipDate shouldn't be a path parameter; pretend it's a similarly defined id
fastify.get('/some/url/:shipDate', { schema } , handler1)
fastify.get('/another/url/:shipDate', { schema }, handler2)
fastify.get('/yetanother/url/:shipDate', { schema }, handler3)
```